File: app/Models/SalesReturn.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class SalesReturn extends Model
{
    use HasFactory;

    protected $fillable = [
        'sales_invoice_id',
        'date',
        'total',
        'reason',
        'created_by',
    ];

    public function salesInvoice()
    {
        return $this->belongsTo(SalesInvoice::class);
    }

    public function items()
    {
        return $this->hasMany(SalesReturnItem::class);
    }

    public function creator()
    {
        return $this->belongsTo(User::class, 'created_by');
    }
}

File: desktop/lib/features/sales/domain/entities/sales_return.dart
import 'package:desktop/features/sales/domain/entities/sales_return_item.dart';

class SalesReturn {
  final String id;
  final String salesInvoiceId;
  final String date;
  final double total;
  final String? reason;
  final int createdBy;
  final List<SalesReturnItem> items;
  final bool isSynced;
  final String? createdAt;
  final String? customerName;

  const SalesReturn({
    required this.id,
    required this.salesInvoiceId,
    required this.date,
    required this.total,
    this.reason,
    required this.createdBy,
    required this.items,
    this.isSynced = false,
    this.createdAt,
    this.customerName,
  });
}
